vendor/bat-native-ledger/src/bat_get_media.cc

```diff
@@ -928,21 +928,54 @@ std::string BatGetMedia::getYoutubeMediaKeyFromUrl(
   return std::string();
 }
 
+// static
 std::string BatGetMedia::getYoutubePublisherKeyFromUrl(const ledger::VisitData& visit_data) {
-  return extractData(visit_data.path + "/", "/channel/", "/");
+  if (visit_data.path.empty()) {
+    return std::string();
+  }
+
+  const std::string id = extractData(visit_data.path + "/", "/channel/", "/");
+
+  if (id.empty()) {
+    return std::string();
+  }
+
+  std::vector<std::string> params = braveledger_bat_helper::split(id, '?');
+
+  return params[0];
```

Comment:
I think we should do a check here to ensure that params has items before accessing this index

Reply:
Is there a case where this will not have at least one item? We pass string in and if there is no `?`, 0 will be passed stirng

Reply:
read this wrong disregard :D